File: 1-patrons-de-creations/abstract-factory/code-typescript/factories/concreteFactories/windowFactory.ts
import { Button } from "../../guiElements/interfaces/button.js";
import { Checkbox } from "../../guiElements/interfaces/checkbox.js";
import { WindowButton } from "../../guiElements/windowElements/windowButton.js";
import { WindowCheckbox } from "../../guiElements/windowElements/windowCheckbox.js";
import { GUIFactory } from "../guiFactory.js";

export class WindowFactory implements GUIFactory {
  createButton(): Button {
    return new WindowButton();
  }

  createCheckbox(): Checkbox {
    return new WindowCheckbox();
  }
}

File: 1-patrons-de-creations/abstract-factory/code-java/factories/concrete_factories/WindowFactory.java
package factories.concrete_factories;

import factories.abstract_factories.GuiFactory;

import gui_elements.interfaces.*;

import gui_elements.window_elements.*;

public class WindowFactory implements GuiFactory {

    public void WindowFactory () {};

     public Button createButton(){
        
        return new WindowButton();
     }

    public Checkbox  createCheckbox() {
        
        return new WindowCkeckbox();
    }
}